### PR TITLE
[MM-22082] - Add focus to server name field in new server modal

### PR DESCRIPTION
**Pull request by nevyangelova**
Before submitting, please confirm you've
 - [x] read and understood our [Contributing Guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md)
 - [x] completed [Mattermost Contributor Agreement](http://www.mattermost.org/mattermost-contributor-agreement/)
 - [x] executed `npm run lint:js` for proper code formatting

Please provide the following information:

**Summary**
By changing the autoFocus to an html property it seems to fix the issue!

**Issue link**
https://mattermost.atlassian.net/browse/MM-22082

### CODE REVIEW

**Comment by nevyangelova** [comment]
@jgilliam17 fixed the issues you mentioned, but on a second initialisation there is a small delay in focus, this is due to the app constantly re-rendering. This is an issue in the desktop app that I am not sure how to fix at this point and therefore can't avoid this delay. But I am happy to start that conversation.

**Comment by jgilliam17** [comment]
> @jgilliam17 fixed the issues you mentioned, but on a second initialisation there is a small delay in focus, this is due to the app constantly re-rendering. This is an issue in the desktop app that I am not sure how to fix at this point and therefore can't avoid this delay. But I am happy to start that conversation.

Thank you @nevyangelova 
I'll take a look tomorrow with the new build.

**Comment by nevyangelova** [comment]
PR Updated. Looks like it works this time with a slight delay on initialisation. That's specific to the popup behaviour, I went through the documentation and found this to be the best working solution in our case. Any other solution I tried either focused solely on the name input and would stop the user from proceeding to the next step or would focus only on first show.

File: src/browser/components/NewTeamModal.jsx [hunk 1]
```diff
@@ -172,10 +173,13 @@ export default class NewTeamModal extends React.Component {
...
                 placeholder='Server Name'
                 onChange={this.handleTeamNameChange}
+                inputRef={(ref) => {
+                  this.teamNameInputRef = ref;
+                }}
                 onClick={(e) => {
                   e.stopPropagation();
                 }}
-                autoFocus={true}
+                autofocus='true'
```
Comment (Willyfrog) [inline src/browser/components/NewTeamModal.jsx]: Shouldn't this be a Boolean?

File: src/browser/components/NewTeamModal.jsx [hunk 2]
```diff
@@ -138,6 +138,7 @@ export default class NewTeamModal extends React.Component {
         className='NewTeamModal'
         show={this.props.show}
         id='newServerModal'
+        onEntered={() => this.teamNameInputRef.focus()}
```
Comment (devinbinnie) [inline src/browser/components/NewTeamModal.jsx]: Just curious how come you need this?
Doesn't `autofocus` do the job?

Comment (nevyangelova) [reply src/browser/components/NewTeamModal.jsx]: @devinbinnie this is to address the issue on second initialisation that @jgilliam17 found, the popup doesn't unmount when you close it and then when you show it again this needs to be set manually :)

**Comment by nevyangelova** [comment]
@esethna is this ready to merge?

**Comment by esethna** [comment]
Yes please merge

**Pull request merged**